### PR TITLE
Jackknife

### PR DESCRIPTION
**Pull request by suvamB**

### CODE REVIEW

File: codes/Jackknife.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import numpy as np
+a=np.loadtxt('0.txt')
```
Comment (leehehehe) [inline codes/Jackknife.py]: This should not be committed

File: codes/Jackknife.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    jack_mean=np.mean(jack_bin)
+    for k in range(M):
+        delta_f=jack_bin[k]-jack_mean
+        delta_f=delta_f**2
+    delta_f=np.sqrt(M/(M-1)*delta_f) 
+    bias= M
+    return delta_f
+
+    
+jackknife(a,10)
```
Comment (leehehehe) [inline codes/Jackknife.py]: Those last 3 lines should be deleted

File: codes/Jackknife.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def jackknife(a,M,):
+    size=np.size(a)
+    i=np.int_(size/M)
+    jack_bin=np.zeros(M)
+    sum_elements=np.sum(a)
+    for k in range(M):
+        jack_bin[k]=sum_elements-np.sum(a[k*i:(k+1)*i])
+        jack_bin[k]/=size
```
Comment (leehehehe) [inline codes/Jackknife.py]: I think that it should be `jack_bin[k] /= i` here, the size of each bin is $i$, not size(a)

File: codes/Jackknife.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    avg_elements=sum_elements/M
```
Comment (leehehehe) [inline codes/Jackknife.py]: What does this line mean?

File: codes/Jackknife.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    
+    jack_mean=np.mean(jack_bin)
+    for k in range(M):
```
Comment (leehehehe) [inline codes/Jackknife.py]: Codes within this for block seem completely wrong to me...

File: codes/Jackknife.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import numpy as np
+a=np.loadtxt('0.txt')
+
+
+
+def jackknife(a,M,):
```
Comment (leehehehe) [inline codes/Jackknife.py]: Actually, here f should be the function of the variance of E, but not E

**Comment by leehehehe** [comment]
Check this: [http://www.hep.fsu.edu/~berg/teach/mcmc08/homework/mcmc_Cv.pdf](url)

**Comment by suvamB** [comment]
the file you sent is not opening 



De: "Honghao" <notifications@github.com> 
À: "leehehehe/phy571_projet" <phy571_projet@noreply.github.com> 
Cc: "suvamB" <suvam.bhattacharya@polytechnique.edu>, "Author" <author@noreply.github.com> 
Envoyé: Mardi 15 Novembre 2016 22:57:07 
Objet: Re: [leehehehe/phy571_projet] Jackknife (#13) 



Check this: http://www.hep.fsu.edu/~berg/teach/mcmc08/homework/mcmc_Cv.pdf

**Comment by leehehehe** [comment]
Et encore: [http://pitp.physics.ubc.ca/confs/sherbrooke/archives/pitpcifar_sherbrooke2008_intro_mc_troyer.pdf](url)
[http://www.helsinki.fi/~rummukai/lectures/montecarlo_oulu/lectures/mc_notes5.pdf](url)

**Comment by leehehehe** [comment]
Since you didn't make any update, I'll close this PL. I've committed a correct version of jackknife.

**Pull request closed without merging**